<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>reports.html</title>
    <link href="assets/style.css" rel="stylesheet" type="text/css"/></head>
  <body onLoad="init()">
    <script>/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */


function toArray(iter) {
    if (iter === null) {
        return null;
    }
    return Array.prototype.slice.call(iter);
}

function find(selector, elem) { // eslint-disable-line no-redeclare
    if (!elem) {
        elem = document;
    }
    return elem.querySelector(selector);
}

function findAll(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return toArray(elem.querySelectorAll(selector));
}

function sortColumn(elem) {
    toggleSortStates(elem);
    const colIndex = toArray(elem.parentNode.childNodes).indexOf(elem);
    let key;
    if (elem.classList.contains('result')) {
        key = keyResult;
    } else if (elem.classList.contains('links')) {
        key = keyLink;
    } else {
        key = keyAlpha;
    }
    sortTable(elem, key(colIndex));
}

function showAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(showExtras);
}

function hideAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(hideExtras);
}

function showExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.remove('collapsed');
    expandcollapse.classList.remove('expander');
    expandcollapse.classList.add('collapser');
}

function hideExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.add('collapsed');
    expandcollapse.classList.remove('collapser');
    expandcollapse.classList.add('expander');
}

function showFilters() {
    let visibleString = getQueryParameter('visible') || 'all';
    visibleString = visibleString.toLowerCase();
    const checkedItems = visibleString.split(',');

    const filterItems = document.getElementsByClassName('filter');
    for (let i = 0; i < filterItems.length; i++) {
        filterItems[i].hidden = false;

        if (visibleString != 'all') {
            filterItems[i].checked = checkedItems.includes(filterItems[i].getAttribute('data-test-result'));
            filterTable(filterItems[i]);
        }
    }
}

function addCollapse() {
    // Add links for show/hide all
    const resulttable = find('table#results-table');
    const showhideall = document.createElement('p');
    showhideall.innerHTML = '<a href="javascript:showAllExtras()">Show all details</a> / ' +
                            '<a href="javascript:hideAllExtras()">Hide all details</a>';
    resulttable.parentElement.insertBefore(showhideall, resulttable);

    // Add show/hide link to each result
    findAll('.col-result').forEach(function(elem) {
        const collapsed = getQueryParameter('collapsed') || 'Passed';
        const extras = elem.parentNode.nextElementSibling;
        const expandcollapse = document.createElement('span');
        if (extras.classList.contains('collapsed')) {
            expandcollapse.classList.add('expander');
        } else if (collapsed.includes(elem.innerHTML)) {
            extras.classList.add('collapsed');
            expandcollapse.classList.add('expander');
        } else {
            expandcollapse.classList.add('collapser');
        }
        elem.appendChild(expandcollapse);

        elem.addEventListener('click', function(event) {
            if (event.currentTarget.parentNode.nextElementSibling.classList.contains('collapsed')) {
                showExtras(event.currentTarget);
            } else {
                hideExtras(event.currentTarget);
            }
        });
    });
}

function getQueryParameter(name) {
    const match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
}

function init () { // eslint-disable-line no-unused-vars
    resetSortHeaders();

    addCollapse();

    showFilters();

    sortColumn(find('.initial-sort'));

    findAll('.sortable').forEach(function(elem) {
        elem.addEventListener('click',
            function() {
                sortColumn(elem);
            }, false);
    });
}

function sortTable(clicked, keyFunc) {
    const rows = findAll('.results-table-row');
    const reversed = !clicked.classList.contains('asc');
    const sortedRows = sort(rows, keyFunc, reversed);
    /* Whole table is removed here because browsers acts much slower
     * when appending existing elements.
     */
    const thead = document.getElementById('results-table-head');
    document.getElementById('results-table').remove();
    const parent = document.createElement('table');
    parent.id = 'results-table';
    parent.appendChild(thead);
    sortedRows.forEach(function(elem) {
        parent.appendChild(elem);
    });
    document.getElementsByTagName('BODY')[0].appendChild(parent);
}

function sort(items, keyFunc, reversed) {
    const sortArray = items.map(function(item, i) {
        return [keyFunc(item), i];
    });

    sortArray.sort(function(a, b) {
        const keyA = a[0];
        const keyB = b[0];

        if (keyA == keyB) return 0;

        if (reversed) {
            return keyA < keyB ? 1 : -1;
        } else {
            return keyA > keyB ? 1 : -1;
        }
    });

    return sortArray.map(function(item) {
        const index = item[1];
        return items[index];
    });
}

function keyAlpha(colIndex) {
    return function(elem) {
        return elem.childNodes[1].childNodes[colIndex].firstChild.data.toLowerCase();
    };
}

function keyLink(colIndex) {
    return function(elem) {
        const dataCell = elem.childNodes[1].childNodes[colIndex].firstChild;
        return dataCell == null ? '' : dataCell.innerText.toLowerCase();
    };
}

function keyResult(colIndex) {
    return function(elem) {
        const strings = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed',
            'Skipped', 'Passed'];
        return strings.indexOf(elem.childNodes[1].childNodes[colIndex].firstChild.data);
    };
}

function resetSortHeaders() {
    findAll('.sort-icon').forEach(function(elem) {
        elem.parentNode.removeChild(elem);
    });
    findAll('.sortable').forEach(function(elem) {
        const icon = document.createElement('div');
        icon.className = 'sort-icon';
        icon.textContent = 'vvv';
        elem.insertBefore(icon, elem.firstChild);
        elem.classList.remove('desc', 'active');
        elem.classList.add('asc', 'inactive');
    });
}

function toggleSortStates(elem) {
    //if active, toggle between asc and desc
    if (elem.classList.contains('active')) {
        elem.classList.toggle('asc');
        elem.classList.toggle('desc');
    }

    //if inactive, reset all other functions and add ascending active
    if (elem.classList.contains('inactive')) {
        resetSortHeaders();
        elem.classList.remove('inactive');
        elem.classList.add('active');
    }
}

function isAllRowsHidden(value) {
    return value.hidden == false;
}

function filterTable(elem) { // eslint-disable-line no-unused-vars
    const outcomeAtt = 'data-test-result';
    const outcome = elem.getAttribute(outcomeAtt);
    const classOutcome = outcome + ' results-table-row';
    const outcomeRows = document.getElementsByClassName(classOutcome);

    for(let i = 0; i < outcomeRows.length; i++){
        outcomeRows[i].hidden = !elem.checked;
    }

    const rows = findAll('.results-table-row').filter(isAllRowsHidden);
    const allRowsHidden = rows.length == 0 ? true : false;
    const notFoundMessage = document.getElementById('not-found-message');
    notFoundMessage.hidden = !allRowsHidden;
}
</script>
    <h1>reports.html</h1>
    <p>Report generated on 02-Jul-2023 at 02:21:10 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a> v3.2.0</p>
    <h2>Summary</h2>
    <p>27 tests ran in 33.13 seconds. </p>
    <p class="filter" hidden="true">(Un)check the boxes to filter the results.</p><input checked="true" class="filter" data-test-result="passed" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="passed">11 passed</span>, <input checked="true" class="filter" data-test-result="skipped" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="skipped">0 skipped</span>, <input checked="true" class="filter" data-test-result="failed" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="failed">16 failed</span>, <input checked="true" class="filter" data-test-result="error" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="error">0 errors</span>, <input checked="true" class="filter" data-test-result="xfailed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xfailed">0 expected failures</span>, <input checked="true" class="filter" data-test-result="xpassed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xpassed">0 unexpected passes</span>
    <h2>Results</h2>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable result initial-sort" col="result">Result</th>
          <th class="sortable" col="name">Test</th>
          <th class="sortable" col="duration">Duration</th>
          <th class="sortable links" col="links">Links</th></tr>
        <tr hidden="true" id="not-found-message">
          <th colspan="4">No results found. Try to check the filters</th></tr></thead>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_with_invalid_users_list</td>
          <td class="col-duration">1.42</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_country.test_count_by_country.TestCountByCountry object at 0x104c113d0&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_country = 4<br/><br/>    def test_count_by_country_with_invalid_users_list(self, api_url, headers, top_country):<br/>        test_users = generate_random_users()<br/>        test_users.append({&#x27;invalid_name&#x27;: &#x27;test&#x27;, &#x27;invalid_value&#x27;: &#x27;test&#x27;})<br/>        response = \<br/>            make_post_request(api_url, test_users, action_type=&quot;CountByCountry&quot;, top=top_country, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_country/test_count_by_country.py:78: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:75 Verifying that the response status code are in [400, 422].
INFO     root:test_count_by_country.py:76 Expected responses: [400, 422]
INFO     root:test_count_by_country.py:77 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_with_invalid_action_type[]</td>
          <td class="col-duration">1.60</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_country.test_count_by_country.TestCountByCountry object at 0x104c12610&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_country = 4, action_type = &#x27;&#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_by_country_with_invalid_action_type(self, api_url, headers, top_country, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_country, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_country/test_count_by_country.py:90: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:87 Verifying that the response status code are in [400, 422].
INFO     root:test_count_by_country.py:88 Expected responses: [400, 422]
INFO     root:test_count_by_country.py:89 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_with_invalid_action_type[ ]</td>
          <td class="col-duration">1.70</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_country.test_count_by_country.TestCountByCountry object at 0x104c12850&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_country = 4, action_type = &#x27; &#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_by_country_with_invalid_action_type(self, api_url, headers, top_country, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_country, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_country/test_count_by_country.py:90: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:87 Verifying that the response status code are in [400, 422].
INFO     root:test_count_by_country.py:88 Expected responses: [400, 422]
INFO     root:test_count_by_country.py:89 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_with_invalid_action_type[invalid_action_type]</td>
          <td class="col-duration">1.49</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_country.test_count_by_country.TestCountByCountry object at 0x104c12b10&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_country = 4<br/>action_type = &#x27;invalid_action_type&#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_by_country_with_invalid_action_type(self, api_url, headers, top_country, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_country, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_country/test_count_by_country.py:90: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:87 Verifying that the response status code are in [400, 422].
INFO     root:test_count_by_country.py:88 Expected responses: [400, 422]
INFO     root:test_count_by_country.py:89 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_without_action_type</td>
          <td class="col-duration">1.24</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_country.test_count_by_country.TestCountByCountry object at 0x104c131d0&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_country = 4<br/><br/>    def test_count_by_country_without_action_type(self, api_url, headers, top_country):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=None, top=top_country, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_country/test_count_by_country.py:101: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:98 Verifying that the response status code are in [400, 422].
INFO     root:test_count_by_country.py:99 Expected responses: [400, 422]
INFO     root:test_count_by_country.py:100 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_with_invalid_users_list</td>
          <td class="col-duration">1.29</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_gender.test_count_by_gender.TestCountByGender object at 0x104c22d10&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_gender = 2<br/><br/>    def test_count_by_gender_with_invalid_users_list(self, api_url, headers, top_gender):<br/>        test_users = generate_random_users()<br/>        test_users.append({&#x27;invalid_name&#x27;: &#x27;test&#x27;, &#x27;invalid_value&#x27;: &#x27;test&#x27;})<br/>        response = make_post_request(api_url, test_users, action_type=&quot;CountByGender&quot;, top=top_gender, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status codes are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_gender/test_count_by_gender.py:81: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:78 Verifying that the response status codes are in [400, 422].
INFO     root:test_count_by_gender.py:79 Expected responses: [400, 422]
INFO     root:test_count_by_gender.py:80 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_with_invalid_action_type[]</td>
          <td class="col-duration">1.46</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_gender.test_count_by_gender.TestCountByGender object at 0x104c23c90&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_gender = 2, action_type = &#x27;&#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_by_gender_with_invalid_action_type(self, api_url, headers, top_gender, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_gender, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status codes are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_gender/test_count_by_gender.py:93: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:90 Verifying that the response status codes are in [400, 422].
INFO     root:test_count_by_gender.py:91 Expected responses: [400, 422]
INFO     root:test_count_by_gender.py:92 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_with_invalid_action_type[ ]</td>
          <td class="col-duration">1.36</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_gender.test_count_by_gender.TestCountByGender object at 0x104c23ed0&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_gender = 2, action_type = &#x27; &#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_by_gender_with_invalid_action_type(self, api_url, headers, top_gender, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_gender, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status codes are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_gender/test_count_by_gender.py:93: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:90 Verifying that the response status codes are in [400, 422].
INFO     root:test_count_by_gender.py:91 Expected responses: [400, 422]
INFO     root:test_count_by_gender.py:92 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_with_invalid_action_type[invalid_action_type]</td>
          <td class="col-duration">1.80</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_gender.test_count_by_gender.TestCountByGender object at 0x104c281d0&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_gender = 2<br/>action_type = &#x27;invalid_action_type&#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_by_gender_with_invalid_action_type(self, api_url, headers, top_gender, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_gender, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status codes are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_gender/test_count_by_gender.py:93: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:90 Verifying that the response status codes are in [400, 422].
INFO     root:test_count_by_gender.py:91 Expected responses: [400, 422]
INFO     root:test_count_by_gender.py:92 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_without_action_type</td>
          <td class="col-duration">1.68</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_by_gender.test_count_by_gender.TestCountByGender object at 0x104c28910&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_gender = 2<br/><br/>    def test_count_by_gender_without_action_type(self, api_url, headers, top_gender):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=None, top=top_gender, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_by_gender/test_count_by_gender.py:104: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:101 Verifying that the response status code are in [400, 422].
INFO     root:test_count_by_gender.py:102 Expected responses: [400, 422]
INFO     root:test_count_by_gender.py:103 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_returns_correct_values[5]</td>
          <td class="col-duration">1.87</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_password_complexity.test_count_password_complexity.TestCountPasswordComplexity object at 0x104c2aa10&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_pw_complex = 3, num_of_users = 5<br/><br/>    @pytest.mark.parametrize(&quot;num_of_users&quot;, [-1, 0, 1, 5])<br/>    def test_count_pw_complexity_returns_correct_values(self, api_url, headers, top_pw_complex, num_of_users):<br/>        test_users = generate_random_users(num_of_users)<br/>        response = make_post_request(<br/>            api_url, test_users, action_type=&quot;CountPasswordComplexity&quot;, top=top_pw_complex, headers=headers<br/>        )<br/>    <br/>        expected_status_code = 200<br/>        logging.info(f&#x27;Expected status code: {expected_status_code}. Actual status code: {response.status_code}&#x27;)<br/>        assert expected_status_code == response.status_code, \<br/>            f&#x27;Error: expected status code was {expected_status_code}, but actual status code was {response.status_code}&#x27;<br/>    <br/>        result = response.json()<br/>    <br/>        if num_of_users &lt;= 0:<br/>            num_users = len(test_users)<br/>            num_results = len(result)<br/>            logging.info(<br/>                f&#x27;Verifying that the number of occurrences (actual: {num_results})&#x27;<br/>                f&#x27; is equal to 0 when there are 0 users (actual: {num_users}).&#x27;<br/>            )<br/>            assert num_users == num_results == 0, \<br/>                f&#x27;Error: Expected zero users and results, but got {num_users} users and {num_results} results.&#x27;<br/>        else:<br/>            expected_length = len(test_users) if top_pw_complex &lt;= 0 or top_pw_complex &gt; len(test_users) \<br/>                else top_pw_complex<br/>    <br/>            if top_pw_complex &lt;= 0 or top_pw_complex &gt; len(test_users):<br/>                log_message = &#x27;Verifying that the number of occurrences is equal to the number of generated users.&#x27;<br/>            else:<br/>                log_message = &#x27;Verifying that the number of occurrences is equal to the specified &quot;top&quot; value.&#x27;<br/>    <br/>            logging.info(log_message)<br/>            logging.info(f&#x27;Expected length of result: {expected_length}&#x27;)<br/>            logging.info(f&#x27;Actual length of result:   {len(result)}&#x27;)<br/>&gt;           assert expected_length == len(result), &#x27;The number of results does not match the expected value.&#x27;<br/><span class="error">E           AssertionError: The number of results does not match the expected value.</span><br/><span class="error">E           assert 3 == 2</span><br/><span class="error">E            +  where 2 = len([{&#x27;name&#x27;: &#x27;lionking&#x27;, &#x27;value&#x27;: 0}, {&#x27;name&#x27;: &#x27;cheryl&#x27;, &#x27;value&#x27;: 0}])</span><br/><br/>tests/count_password_complexity/test_count_password_complexity.py:59: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:32 Expected status code: 200. Actual status code: 200
INFO     root:test_count_password_complexity.py:56 Verifying that the number of occurrences is equal to the specified &quot;top&quot; value.
INFO     root:test_count_password_complexity.py:57 Expected length of result: 3
INFO     root:test_count_password_complexity.py:58 Actual length of result:   2<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_with_invalid_users_list</td>
          <td class="col-duration">1.52</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_password_complexity.test_count_password_complexity.TestCountPasswordComplexity object at 0x104c2b090&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_pw_complex = 3<br/><br/>    def test_count_pw_complexity_with_invalid_users_list(self, api_url, headers, top_pw_complex):<br/>        test_users = generate_random_users()<br/>        test_users.append({&#x27;invalid_name&#x27;: &#x27;test&#x27;, &#x27;invalid_value&#x27;: &#x27;test&#x27;})<br/>        response = make_post_request(<br/>            api_url, test_users, action_type=&quot;CountPasswordComplexity&quot;, top=top_pw_complex, headers=headers<br/>        )<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_password_complexity/test_count_password_complexity.py:82: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:79 Verifying that the response status code are in [400, 422].
INFO     root:test_count_password_complexity.py:80 Expected responses: [400, 422]
INFO     root:test_count_password_complexity.py:81 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_with_invalid_action_type[]</td>
          <td class="col-duration">1.51</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_password_complexity.test_count_password_complexity.TestCountPasswordComplexity object at 0x104c38050&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_pw_complex = 3, action_type = &#x27;&#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_pw_complexity_with_invalid_action_type(self, api_url, headers, top_pw_complex, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_pw_complex, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_password_complexity/test_count_password_complexity.py:94: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:91 Verifying that the response status code are in [400, 422].
INFO     root:test_count_password_complexity.py:92 Expected responses: [400, 422]
INFO     root:test_count_password_complexity.py:93 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_with_invalid_action_type[ ]</td>
          <td class="col-duration">1.39</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_password_complexity.test_count_password_complexity.TestCountPasswordComplexity object at 0x104c38250&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_pw_complex = 3, action_type = &#x27; &#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_pw_complexity_with_invalid_action_type(self, api_url, headers, top_pw_complex, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_pw_complex, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_password_complexity/test_count_password_complexity.py:94: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:91 Verifying that the response status code are in [400, 422].
INFO     root:test_count_password_complexity.py:92 Expected responses: [400, 422]
INFO     root:test_count_password_complexity.py:93 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_with_invalid_action_type[invalid_action_type]</td>
          <td class="col-duration">1.47</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_password_complexity.test_count_password_complexity.TestCountPasswordComplexity object at 0x104c38510&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_pw_complex = 3<br/>action_type = &#x27;invalid_action_type&#x27;<br/><br/>    @pytest.mark.parametrize(&quot;action_type&quot;, [&quot;&quot;, &quot; &quot;, &quot;invalid_action_type&quot;])<br/>    def test_count_pw_complexity_with_invalid_action_type(self, api_url, headers, top_pw_complex, action_type):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=action_type, top=top_pw_complex, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_password_complexity/test_count_password_complexity.py:94: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:91 Verifying that the response status code are in [400, 422].
INFO     root:test_count_password_complexity.py:92 Expected responses: [400, 422]
INFO     root:test_count_password_complexity.py:93 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_without_action_type</td>
          <td class="col-duration">1.60</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;tests.count_password_complexity.test_count_password_complexity.TestCountPasswordComplexity object at 0x104c38c50&gt;<br/>api_url = &#x27;https://census-toy.nceng.net/prod/toy-census&#x27;<br/>headers = {&#x27;Accept&#x27;: &#x27;application/json&#x27;, &#x27;Content-Type&#x27;: &#x27;application/json&#x27;}, top_pw_complex = 3<br/><br/>    def test_count_pw_complexity_without_action_type(self, api_url, headers, top_pw_complex):<br/>        test_users = generate_random_users()<br/>        response = make_post_request(api_url, test_users, action_type=None, top=top_pw_complex, headers=headers)<br/>    <br/>        expected_response_codes = [400, 422]<br/>        logging.info(f&#x27;Verifying that the response status code are in {expected_response_codes}.&#x27;)<br/>        logging.info(f&#x27;Expected responses: {expected_response_codes}&#x27;)<br/>        logging.info(f&#x27;Actual response:    {response.status_code}&#x27;)<br/>&gt;       assert response.status_code in expected_response_codes, \<br/>            f&#x27;Error: Expected response code {expected_response_codes}, but received {response.status_code}&#x27;<br/><span class="error">E       AssertionError: Error: Expected response code [400, 422], but received 200</span><br/><span class="error">E       assert 200 in [400, 422]</span><br/><span class="error">E        +  where 200 = &lt;Response [200]&gt;.status_code</span><br/><br/>tests/count_password_complexity/test_count_password_complexity.py:105: AssertionError<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:102 Verifying that the response status code are in [400, 422].
INFO     root:test_count_password_complexity.py:103 Expected responses: [400, 422]
INFO     root:test_count_password_complexity.py:104 Actual response:    200<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_returns_correct_values[-1]</td>
          <td class="col-duration">0.51</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_country.py:39 Verifying that the number of occurrences (actual: 0) is equal to 0 when there are 0 users (actual: 0).
INFO     root:test_count_by_country.py:58 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_country.py:60 Expected result: []
INFO     root:test_count_by_country.py:61 Actual result:   []<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_returns_correct_values[0]</td>
          <td class="col-duration">0.56</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_country.py:39 Verifying that the number of occurrences (actual: 0) is equal to 0 when there are 0 users (actual: 0).
INFO     root:test_count_by_country.py:58 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_country.py:60 Expected result: []
INFO     root:test_count_by_country.py:61 Actual result:   []<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_returns_correct_values[1]</td>
          <td class="col-duration">0.68</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_country.py:53 Verifying that the number of occurrences is equal to the number of generated users.
INFO     root:test_count_by_country.py:54 Expected length of result: 1
INFO     root:test_count_by_country.py:55 Actual length of result:   1
INFO     root:test_count_by_country.py:58 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_country.py:60 Expected result: [{&#x27;name&#x27;: &#x27;AU&#x27;, &#x27;value&#x27;: 1}]
INFO     root:test_count_by_country.py:61 Actual result:   [{&#x27;name&#x27;: &#x27;AU&#x27;, &#x27;value&#x27;: 1}]<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_country/test_count_by_country.py::TestCountByCountry::test_count_by_country_returns_correct_values[5]</td>
          <td class="col-duration">1.50</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_country.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_country.py:53 Verifying that the number of occurrences is equal to the specified &quot;top&quot; value.
INFO     root:test_count_by_country.py:54 Expected length of result: 4
INFO     root:test_count_by_country.py:55 Actual length of result:   4
INFO     root:test_count_by_country.py:58 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_country.py:60 Expected result: [{&#x27;name&#x27;: &#x27;IN&#x27;, &#x27;value&#x27;: 1}, {&#x27;name&#x27;: &#x27;AU&#x27;, &#x27;value&#x27;: 1}, {&#x27;name&#x27;: &#x27;BR&#x27;, &#x27;value&#x27;: 1}, {&#x27;name&#x27;: &#x27;UA&#x27;, &#x27;value&#x27;: 1}]
INFO     root:test_count_by_country.py:61 Actual result:   [{&#x27;name&#x27;: &#x27;IN&#x27;, &#x27;value&#x27;: 1}, {&#x27;name&#x27;: &#x27;AU&#x27;, &#x27;value&#x27;: 1}, {&#x27;name&#x27;: &#x27;BR&#x27;, &#x27;value&#x27;: 1}, {&#x27;name&#x27;: &#x27;UA&#x27;, &#x27;value&#x27;: 1}]<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_returns_correct_values[-1]</td>
          <td class="col-duration">0.35</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_gender.py:39 Verifying that the number of occurrences (actual: 0) is equal to 0 when there are 0 users (actual: 0).
INFO     root:test_count_by_gender.py:62 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_gender.py:64 Expected result: []
INFO     root:test_count_by_gender.py:65 Actual result:   []<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_returns_correct_values[0]</td>
          <td class="col-duration">0.55</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_gender.py:39 Verifying that the number of occurrences (actual: 0) is equal to 0 when there are 0 users (actual: 0).
INFO     root:test_count_by_gender.py:62 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_gender.py:64 Expected result: []
INFO     root:test_count_by_gender.py:65 Actual result:   []<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_returns_correct_values[1]</td>
          <td class="col-duration">0.71</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_gender.py:49 Verifying that the result contains 1 entries.
INFO     root:test_count_by_gender.py:50 Expected length of result: 1
INFO     root:test_count_by_gender.py:51 Actual length of result:   1
INFO     root:test_count_by_gender.py:56 Verifying that the sum of values equals the number of generated users.
INFO     root:test_count_by_gender.py:57 Expected sum of values: 1
INFO     root:test_count_by_gender.py:58 Actual sum of values:   1
INFO     root:test_count_by_gender.py:62 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_gender.py:64 Expected result: [{&#x27;name&#x27;: &#x27;female&#x27;, &#x27;value&#x27;: 1}]
INFO     root:test_count_by_gender.py:65 Actual result:   [{&#x27;name&#x27;: &#x27;female&#x27;, &#x27;value&#x27;: 1}]<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_by_gender/test_count_by_gender.py::TestCountByGender::test_count_by_gender_returns_correct_values[5]</td>
          <td class="col-duration">1.49</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_by_gender.py:30 Expected status code: 200. Actual status code: 200
INFO     root:test_count_by_gender.py:49 Verifying that the result contains 2 entries.
INFO     root:test_count_by_gender.py:50 Expected length of result: 2
INFO     root:test_count_by_gender.py:51 Actual length of result:   2
INFO     root:test_count_by_gender.py:56 Verifying that the sum of values equals the number of generated users.
INFO     root:test_count_by_gender.py:57 Expected sum of values: 5
INFO     root:test_count_by_gender.py:58 Actual sum of values:   5
INFO     root:test_count_by_gender.py:62 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_by_gender.py:64 Expected result: [{&#x27;name&#x27;: &#x27;female&#x27;, &#x27;value&#x27;: 4}, {&#x27;name&#x27;: &#x27;male&#x27;, &#x27;value&#x27;: 1}]
INFO     root:test_count_by_gender.py:65 Actual result:   [{&#x27;name&#x27;: &#x27;female&#x27;, &#x27;value&#x27;: 4}, {&#x27;name&#x27;: &#x27;male&#x27;, &#x27;value&#x27;: 1}]<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_returns_correct_values[-1]</td>
          <td class="col-duration">0.54</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:32 Expected status code: 200. Actual status code: 200
INFO     root:test_count_password_complexity.py:41 Verifying that the number of occurrences (actual: 0) is equal to 0 when there are 0 users (actual: 0).
INFO     root:test_count_password_complexity.py:61 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_password_complexity.py:63 Expected result: []
INFO     root:test_count_password_complexity.py:64 Actual result:   []<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_returns_correct_values[0]</td>
          <td class="col-duration">0.72</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:32 Expected status code: 200. Actual status code: 200
INFO     root:test_count_password_complexity.py:41 Verifying that the number of occurrences (actual: 0) is equal to 0 when there are 0 users (actual: 0).
INFO     root:test_count_password_complexity.py:61 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_password_complexity.py:63 Expected result: []
INFO     root:test_count_password_complexity.py:64 Actual result:   []<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/count_password_complexity/test_count_password_complexity.py::TestCountPasswordComplexity::test_count_pw_complexity_returns_correct_values[1]</td>
          <td class="col-duration">0.78</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> -------------------------------Captured log call-------------------------------- <br/>INFO     root:test_count_password_complexity.py:32 Expected status code: 200. Actual status code: 200
INFO     root:test_count_password_complexity.py:56 Verifying that the number of occurrences is equal to the number of generated users.
INFO     root:test_count_password_complexity.py:57 Expected length of result: 1
INFO     root:test_count_password_complexity.py:58 Actual length of result:   1
INFO     root:test_count_password_complexity.py:61 Verifying that the return values are ordered from highest to lowest.
INFO     root:test_count_password_complexity.py:63 Expected result: [{&#x27;name&#x27;: &#x27;peanut&#x27;, &#x27;value&#x27;: 0}]
INFO     root:test_count_password_complexity.py:64 Actual result:   [{&#x27;name&#x27;: &#x27;peanut&#x27;, &#x27;value&#x27;: 0}]<br/></div></td></tr></tbody></table></body></html>